src/app/story/directives/player.component.ts

```diff
@@ -1,122 +1,157 @@
+      let img = story.images.find(i => !i.isDestroy);
+      if (img) { this.imageUrl = img.enclosureHref; }
 
-  get safeUrl() {
-    return this.sanitizer.bypassSecurityTrustResourceUrl(this.embeddableUrl);
+      // TODO: figure out the correct audio version to use
+      let version = story.versions.find(v => !v.isDestroy);
+      if (version) {
+        version.loadRelated('files').subscribe(() => {
+          let audio = version.files.find(f => !f.isDestroy);
+          if (audio) { this.audioUrl = audio.enclosureHref; }
+        });
+      }
+    });
   }
 
-  iframeHtml(width: string, height: string) {
-    const url = this.embeddableUrl;
-    return `<iframe frameborder="0" height="${height}" width="${width}" src="${url}"></iframe>`;
+  fromFeeder(story: StoryModel, dist: DistributionModel) {
+    dist.loadRelated('podcast').subscribe(() => {
+      if (dist.podcast && dist.podcast.publishedUrl) {
+        this.feedUrl = dist.podcast.publishedUrl;
+      } else {
+        this.loadError = 'Error - unable to find the public URL of your podcast!';
+      }
+    });
+    story.loadRelated('distributions').subscribe(() => {
+      let storyDistribution = story.distributions.find(d => d.kind === 'episode');
+      if (storyDistribution) {
+        storyDistribution.loadRelated('episode').subscribe(() => {
+          if (storyDistribution.episode && storyDistribution.episode.guid) {
+            this.episodeGuid = storyDistribution.episode.guid;
+          } else {
+            this.loadError = 'Error - unable to find the guid of this podcast episode!';
+          }
+        });
+      } else {
+        this.loadError = 'Error - no podcast episode set for this story!';
+      }
+    });
   }
 
-  copy(el: any) {
-    const inp = this.inputEl.nativeElement;
-    if (inp && inp.select) {
-      inp.select();
-
-      try {
-        document.execCommand('copy');
-        inp.blur();
-        el.innerHTML = 'Copied';
-      } catch (err) {
-        /// alert('please press Ctrl/Cmd+C to copy');
-      }
+  buildPlayer(previewFeeder: boolean, copyFeeder: boolean) {
+    let cms: string[] = [];
+    cms.push(`tt=${this.encode(this.title)}`);
+    cms.push(`ts=${this.encode(this.subtitle)}`);
+    cms.push(`ua=${this.encode(this.audioUrl)}`);
+    cms.push(`ui=${this.encode(this.imageUrl)}`);
+    cms.push(`us=${this.encode(this.subscriptionUrl)}`);
+    let cmsUrl = `${Env.PLAY_HOST}/e?${cms.join('&')}`;
+
+    let feeder: string[] = [];
+    feeder.push(`uf=${this.encode(this.feedUrl)}`);
+    feeder.push(`ge=${this.encode(this.episodeGuid)}`);
+    let feederUrl = `${Env.PLAY_HOST}/e?${feeder.join('&')}`;
+
+    // only change the safe resource when the url actually changes
+    let newPreview = previewFeeder ? feederUrl : cmsUrl;
+    if (this.previewUrl !== newPreview) {
+      this.previewUrl = newPreview;
+      this.previewSafe = this.sanitizer.bypassSecurityTrustResourceUrl(newPreview);
```

Comment:
Turns out, every time you get `bypassSecurityTrustResourceUrl`, you get a new/unique value.  So calling this every time causes the iframe to flicker like crazy due to change detection.